### PR TITLE
InfluxDB: annotations migration to backend

### PR DESCRIPTION
**Pull request by joey-grafana**
**What this PR does / why we need it**:
Part of https://github.com/grafana/grafana/issues/43076

This PR contains the changes needed to migrate annotations to the backend.

**Test scenarios**:

1. Run make devenv sources=influxdb,influxdb1,grafana grafana_version=8.2.4 (or other version before changes)
2. Open up http://localhost:3000 && http://127.0.0.1:3001 (to test older 8.2.4 version)
3. Select gdev-influxdb-influxql
    - create a dashboard
    - create an annotation query in your dashboard e.g. select * from logs where time >= now() - 30m and time <= now() (be careful with the - 30m above as setting a value too big will potentially return tens of thousands of results. Maybe a smaller time like - 10m to start and increment until you get a number of results you are happy with)
    - add a time series panel hooked up to influxdb
    - set time range to last 3 hours or so (at least enough to cover time range set on annotation query)
    - go back to your dashboard and toggle on the annotation query -> you should see annotations in your dashboard
    - go back to your panel in edit mode -> you should see annotations
4. Select gdev-influxdb1-influxql
    - Should behave like `gdev-influxdb-influxql`

PS: make sure 
[feature_toggles]
enable = influxdbBackendMigration

### CODE REVIEW

File: public/app/plugins/datasource/influxdb/response_parser.ts [hunk 1]
```diff
@@ -84,6 +85,72 @@ export default class ResponseParser {
...
+      });
+
+      each(table.rows, (value) => {
+        const data = {
+          annotation: options.annotation,
+          time: +new Date(value[timeCol]),
+          title: value[titleCol],
+          timeEnd: value[timeEndCol],
+          // Remove empty values, then split in different tags for comma separated values
+          tags: flatten(
```
Comment (yesoreyeram) [inline public/app/plugins/datasource/influxdb/response_parser.ts:135]: Field mapping doesn't seems to work correctly after migration.  Example For the query `SELECT time,host,message,path from logs where time >= now() - 30m and time <= now()`,  the tags are not working as expected after migration.

### Query:
image

### Before migration:
image

### After migration:

image

File: public/app/plugins/datasource/influxdb/specs/response_parser.test.ts [hunk 2]
```diff
@@ -282,4 +296,97 @@ describe('influxdb response parser', () => {
...
+      instanceSettings: { url: 'url', name: 'influxDb' },
+    };
+
+    const fetchMock = jest.spyOn(backendSrv, 'fetch');
+
+    const queryOptions: any = {
+      annotation: {
+        name: 'Anno',
+        query: 'select * from logs where time >= now() - 15m and time <= now()',
+        textColumn: 'textColumn',
```
Comment (yesoreyeram) [inline public/app/plugins/datasource/influxdb/specs/response_parser.test.ts:311]: Can we add tests for field migration as well? especially for tags?

**Review by yesoreyeram — CHANGES_REQUESTED** [review]

File: public/app/plugins/datasource/influxdb/response_parser.ts (hunk 1 above)
Comment (yesoreyeram) [reply public/app/plugins/datasource/influxdb/response_parser.ts:135]: Also there are chances that user have multiple fields as tags.

image
image

Comment (joey-grafana) [reply public/app/plugins/datasource/influxdb/response_parser.ts:135]: Screenshot 2022-03-17 at 12 06 04

**Review by yesoreyeram — APPROVED** [review]
LGTM

**Pull request merged**